src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -163,24 +165,39 @@ public static double BitIncrement(double x)
             return BitConverter.Int64BitsToDouble(bits);
         }
 
-        public static unsafe double CopySign(double x, double y)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double CopySign(double x, double y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
-
-            long xbits = BitConverter.DoubleToInt64Bits(x);
-            long ybits = BitConverter.DoubleToInt64Bits(y);
+            if (Sse.IsSupported || AdvSimd.IsSupported)
+            {
+                var xvec = Vector128.CreateScalarUnsafe(x);
```

Comment:
Nit: This should use explicit type instead of `var` according to the coding convention.